src/lib/generatorFormParameters.ts

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+export type FormParameters = {
+  clientId: string;
+  clientName: string;
+  clientUri: string;
+  redirectUris: string[];
+  useRefreshTokens: boolean;
+  logoUri: string;
+  tosUri: string;
+  policyUri: string;
+  contact: string;
+  applicationType: "web" | "native";
+  requireAuthTime?: boolean;
+  defaultMaxAge?: number;
+};
+
+/**
+ * Return the FormParameters typed key for the string key or `undefined` if not present.
+ * If the key has the form <key name>.<array field index>, the second return value is the index.
+ * Needed for typescript type-safety.
+ * TODO: is there a nicer way to do this?
+ *
+ * @param key The FormParameters key as string.
+ * @returns `[<FormParameters key | undefined>, <index | undefined>]`
+ */
+export function getFormParametersKey(
+  key: string
+): [keyof FormParameters | undefined, number | undefined] {
+  const [fieldName, index] = key.split(".");
+  if (index !== undefined && !Number.isInteger(Number(index))) {
+    return [undefined, undefined];
+  }
+
+  let field: keyof FormParameters | undefined;
+
+  switch (fieldName) {
```

Comment:
Splitting the key between an explicit name and index will allow stricter type checking on the field name, which could be directly typed as `keyOf FormParameter`. This means the default case should not happen, and an error should be thrown otherwise.